pkg/discovery/util/node_util.go

```diff
@@ -30,13 +46,33 @@ func GetNodeIPForMonitor(node *api.Node, source types.MonitoringSource) (string,
 		if ip != "" {
 			return ip, nil
 		}
-		return "", fmt.Errorf("Node %v has no valid hostname and/or IP address: %v %v", node.Name, hostname, ip)
+		return "", fmt.Errorf("node %v has no valid hostname and/or IP address: %v %v", node.Name, hostname, ip)
 	default:
-		return "", errors.New("Unsupported monitoring source or monitoring source not provided")
+		return "", errors.New("unsupported monitoring source or monitoring source not provided")
 	}
 }
 
-// Check if a node is in Ready status.
+func GetNodeOSArch(node *api.Node) (os string, arch string) {
+	os = "unknown"
+	arch = "unknown"
+	labelsMap := node.ObjectMeta.Labels
+	if len(labelsMap) == 0 {
+		return
+	}
+	if val, ok := labelsMap[NodeLabelOS]; ok {
+		os = val
+	} else if val, ok := labelsMap[NodeLabelOSBeta]; ok {
+		os = val
+	}
+	if val, ok := labelsMap[NodeLabelArch]; ok {
+		arch = val
+	} else if val, ok := labelsMap[NodeLabelArchBeta]; ok {
+		arch = val
+	}
+	return
```

Comment:
nit: Will it be helpful to log all the labels in case our predefined label prefix missed? So that we can have more information in case there're other labels used (other than just unknown)

Reply:
Addressed.